pyresample/geometry.py

```diff
@@ -1111,133 +1118,155 @@ def get_lonlat(self, row, col):
+        if chunks is None:
+            chunks = CHUNK_SIZE  # FUTURE: Use a global config object instead
+        return self.get_proj_vectors(dtype=dtype, chunks=chunks)
+
+    def _get_proj_vectors(self, dtype=None, check_rotation=True, chunks=None):
+        """Helper for getting 1D projection coordinates."""
+        x_kwargs = {}
+        y_kwargs = {}
+
+        if chunks is not None and not isinstance(chunks, int):
             y_chunks = chunks[0]
             x_chunks = chunks[1]
         else:
-            y_chunks = chunks
-            x_chunks = chunks
+            y_chunks = x_chunks = chunks
 
-        target_x = da.arange(self.x_size, chunks=x_chunks, dtype=dtype) * \
-            self.pixel_size_x + self.pixel_upper_left[0]
-        target_y = da.arange(self.y_size, chunks=y_chunks, dtype=dtype) * - \
-            self.pixel_size_y + self.pixel_upper_left[1]
+        if x_chunks is not None or y_chunks is not None:
+            # use dask functions instead of numpy
+            from dask.array import arange
+            x_kwargs = {'chunks': x_chunks}
+            y_kwargs = {'chunks': y_chunks}
+        else:
+            arange = np.arange
+        if check_rotation and self.rotation != 0:
+            warnings.warn("Projection vectors will not be accurate because rotation is not 0", RuntimeWarning)
+        if dtype is None:
+            dtype = self.dtype
+        x_kwargs['dtype'] = dtype
+        y_kwargs['dtype'] = dtype
+
+        target_x = arange(self.x_size, **x_kwargs) * self.pixel_size_x + self.pixel_upper_left[0]
+        target_y = arange(self.y_size, **y_kwargs) * -self.pixel_size_y + self.pixel_upper_left[1]
         return target_x, target_y
 
-    def get_proj_coords_dask(self, chunks=CHUNK_SIZE, dtype=None):
-        # TODO: Add rotation
-        import dask.array as da
-        target_x, target_y = self.get_proj_vectors_dask(chunks, dtype)
-        return da.meshgrid(target_x, target_y)
+    def get_proj_vectors(self, dtype=None, chunks=None):
+        """Calculate 1D projection coordinates for the X and Y dimension.
 
-    def get_proj_coords(self, data_slice=None, cache=False, dtype=None):
+        Parameters
+        ----------
+        dtype : numpy.dtype
+            Numpy data type for the returned arrays
+        chunks : int or tuple
+            Return dask arrays with the chunk size specified. If this is a
+            tuple then the first element is the Y array's chunk size and the
+            second is the X array's chunk size.
+
+        Returns
+        -------
+        tuple: (X, Y) where X and Y are 1-dimensional numpy arrays
+
+        The data type of the returned arrays can be controlled with the
+        `dtype` keyword argument. If `chunks` is provided then dask arrays
+        are returned instead.
+
+        """
+        return self._get_proj_vectors(dtype=dtype, chunks=chunks)
```

Comment:
Why use a private function here ?

Reply:
The `_get_proj_vectors` method is used in the `get_proj_coords` method. The main difference is whether or not rotation is supported. With the vectors, like in this method, the rotation is not supported. With the `coords` method the rotation is handled internally. I couldn't figure out how to handle the rotation with just the vectors and the existing code by @loreclem.

So the private method is to hide the `check_rotation` keyword argument from the user.

Reply:
ok